File: news-api/src/main/java/com/news/searchinfo/model/SearchInfo.java
package com.news.searchinfo.model;


import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.ToString;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Builder
@AllArgsConstructor
@ToString
public class SearchInfo {

    private final int id;

    private final int searchCount;

    private final String query;

    private final LocalDateTime eventDateTime;

    public static SearchInfo create(String query, LocalDateTime eventDateTime){
        return SearchInfo.builder()
                .query(query)
                .searchCount(1)
                .eventDateTime(eventDateTime)
                .build();
    }

    public static List<SearchInfo> create(List<String> queryList, LocalDateTime eventDateTime){
        return queryList.stream()
                .map(query -> SearchInfo.create(query, eventDateTime))
                .toList();
    }
}

File: news-api/src/main/kotlin/com/search/domain/searchinfo/model/SearchInfo.kt
package com.search.domain.searchinfo.model

import java.time.LocalDateTime

class SearchInfo private constructor(
        val id: Int? = null,
        val searchCount: Int,
        val query: String,
        val eventDateTime: LocalDateTime
) { companion object{
        @JvmStatic
        fun create(query: String, eventDateTime: LocalDateTime): SearchInfo {
            require(query.isNotBlank()) { "Query cannot be blank" }
            return SearchInfo(
                    id = null,
                    searchCount = 1,
                    query = query,
                    eventDateTime = eventDateTime
                )
            }
        @JvmStatic
        fun createList(queryList: List<String>, eventDateTime: LocalDateTime): List<SearchInfo> {
            return queryList.map { query -> create(query, eventDateTime) }
        }
    }
}

